cpp/ql/src/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -202,49 +202,39 @@ private class ArrayContent extends Content, TArrayContent {
   override string toString() { result = "array content" }
 }
 
-private predicate fieldStoreStepNoChi(Node node1, FieldContent f, PostUpdateNode node2) {
-  exists(StoreInstruction store, Class c |
-    store = node2.asInstruction() and
+/** A store step from the value of a store instruction to the "innermost" field of the destination. */
+private predicate instrToFieldNodeStoreStepNoChi(Node node1, FieldContent f, FieldNode node2) {
+  exists(StoreInstruction store |
+    not exists(ChiInstruction chi | chi.getPartial() = store) and
+    node2 = getFieldNodeForFieldInstruction(store.getDestinationAddress()) and
     store.getSourceValue() = node1.asInstruction() and
-    getWrittenField(store, f.(FieldContent).getAField(), c) and
-    f.hasOffset(c, _, _)
+    getField(f, node2.getField())
   )
 }
 
-private FieldAddressInstruction getFieldInstruction(Instruction instr) {
-  result = instr or
-  result = instr.(CopyValueInstruction).getUnary()
-}
-
-pragma[noinline]
-private predicate getWrittenField(Instruction instr, Field f, Class c) {
-  exists(FieldAddressInstruction fa |
-    fa =
-      getFieldInstruction([instr.(StoreInstruction).getDestinationAddress(),
-            instr.(WriteSideEffectInstruction).getDestinationAddress()]) and
-    f = fa.getField() and
-    c = f.getDeclaringType()
-  )
-}
-
-private predicate fieldStoreStepChi(Node node1, FieldContent f, PostUpdateNode node2) {
-  exists(StoreInstruction store, ChiInstruction chi |
+/**
+ * A store step from a `StoreInstruction` or `WriteSideEffectInstruction` to the "depest" field
+ * of the destination
+ */
```

Comment:
Does "deepest" here mean the same thing as "innermost" above? Also, this comment should say something about chi nodes.

Reply:
Yes, "deepest" here means "innermost". I'll make that consistent across the comments so that they all say "innermost". Also a good point about the Chi instruction. I will clarify that!

Reply:
Fixed in df5ae43cf463e309b3ab735d86149919b3c45751.